analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInGenericClass.cs

```diff
@@ -44,18 +45,19 @@ public sealed class StaticFieldInGenericClass : SonarDiagnosticAnalyzer
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var classDeclaration = (ClassDeclarationSyntax)c.Node;
+                    var typeDeclaration = (TypeDeclarationSyntax)c.Node;
 
-                    if (classDeclaration.TypeParameterList == null
-                        || classDeclaration.TypeParameterList.Parameters.Count < 1)
+                    if (c.ContainingSymbol.Kind != SymbolKind.NamedType
+                        || typeDeclaration.TypeParameterList == null
+                        || typeDeclaration.TypeParameterList.Parameters.Count < 1)
                     {
                         return;
                     }
 
                     var typeParameterNames =
-                        classDeclaration.TypeParameterList.Parameters.Select(p => p.Identifier.ToString()).ToList();
+                        typeDeclaration.TypeParameterList.Parameters.Select(p => p.Identifier.ToString()).ToList();
```

Comment:
pedantic: this ca go on the same line with the var declaration (maximum line length by convention is 200)